### ISSUE TITLE
fedora-34 - dnf update - qubes_hooks.py error

### ISSUE THREAD

**Issue by brendanhoar** [QubesOS/qubes-issues#6993]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release
R4.0 up-to-date w/ current-testing

### Brief summary
Performed: sudo dnf upgrade --enablerepo=qubes-vm-*-current-testing
Result was mostly successful, with the following error towards the end:
```
 Verifying        : wget-1.21.2-1.fc34.x86_64                                                                                                                                                                                                                                                                      71/77 
  Verifying        : wget-1.21.1-3.fc34.x86_64                                                                                                                                                                                                                                                                        72/77 
  Verifying        : kernel-5.14.10-200.fc34.x86_64                                                                                                                                                                                                                                                                   73/77 
  Verifying        : kernel-core-5.14.10-200.fc34.x86_64                                                                                                                                                                                                                                                              74/77 
  Verifying        : kernel-debug-devel-5.14.10-200.fc34.x86_64                                                                                                                                                                                                                                                       75/77 
  Verifying        : kernel-devel-5.14.10-200.fc34.x86_64                                                                                                                                                                                                                                                             76/77 
  Verifying        : kernel-modules-5.14.10-200.fc34.x86_64                                                                                                                                                                                                                                                           77/77 
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/dnf/plugin.py", line 104, in _caller
    getattr(plugin, method)()
  File "/usr/lib/python3.9/site-packages/dnf-plugins/qubes-hooks.py", line 78, in transaction
    updates.discard(item.pkg)
  File "/usr/lib/python3.9/site-packages/dnf/db/history.py", line 157, in pkg
    return self._swdb.rpm._swdb_ti_pkg[self._item]
KeyError: <libdnf.transaction.TransactionItem; proxy of <Swig Object of type 'libdnf::TransactionItemPtr *' at 0x74ff5165c270> >


Upgraded:
  babl-0.1.88-1.fc34.x86_64                        bluez-5.62-1.fc34.x86_64                            bluez-cups-5.62-1.fc34.x86_64                       bluez-libs-5.62-1.fc34.x86_64                     bluez-obexd-5.62-1.fc34.x86_64                     gegl04-0.4.32-1.fc34.x86_64                              
  gnome-online-accounts-3.40.1-1.fc34.x86_64       iwl100-firmware-39.31.5.1-125.fc34.noarch           iwl1000-firmware-1:39.31.5.1-125.fc34.noarch        iwl105-firmware-18.168.6.1-125.fc34.noarch        iwl135-firmware-18.168.6.1-125.fc34.noarch         iwl2000-firmware-18.168.6.1-125.fc34.noarch              
  iwl2030-firmware-18.168.6.1-125.fc34.noarch      iwl3160-firmware-1:25.30.13.0-125.fc34.noarch       iwl3945-firmware-15.32.2.9-125.fc34.noarch          iwl4965-firmware-228.61.2.24-125.fc34.noarch      iwl5000-firmware-8.83.5.1_1-125.fc34.noarch        iwl5150-firmware-8.24.2.2-125.fc34.noarch                
  iwl6000-firmware-9.221.4.1-125.fc34.noarch       iwl6000g2a-firmware-18.168.6.1-125.fc34.noarch      iwl6000g2b-firmware-18.168.6.1-125.fc34.noarch      iwl6050-firmware-41.28.5.1-125.fc34.noarch        iwl7260-firmware-1:25.30.13.0-125.fc34.noarch      libertas-usb8388-firmware-2:20210919-125.fc34.noarch     
  linux-firmware-20210919-125.fc34.noarch          linux-firmware-whence-20210919-125.fc34.noarch      thunderbird-91.2.0-1.fc34.x86_64                    thunderbird-librnp-rnp-91.2.0-1.fc34.x86_64       vim-common-2:8.2.3512-1.fc34.x86_64                vim-enhanced-2:8.2.3512-1.fc34.x86_64                    
  vim-filesystem-2:8.2.3512-1.fc34.noarch          vim-minimal-2:8.2.3512-1.fc34.x86_64                wget-1.21.2-1.fc34.x86_64                          
Installed:
  iwlax2xx-firmware-20210919-125.fc34.noarch               kernel-5.14.13-200.fc34.x86_64               kernel-core-5.14.13-200.fc34.x86_64               kernel-debug-devel-5.14.13-200.fc34.x86_64               kernel-devel-5.14.13-200.fc34.x86_64               kernel-modules-5.14.13-200.fc34.x86_64              
Removed:
  kernel-5.14.10-200.fc34.x86_64                          kernel-core-5.14.10-200.fc34.x86_64                          kernel-debug-devel-5.14.10-200.fc34.x86_64                          kernel-devel-5.14.10-200.fc34.x86_64                          kernel-modules-5.14.10-200.fc34.x86_64                         

Complete!
[user@fedora-34 ~]$
```




### Steps to reproduce
Start fedora-34 template
Open Terminal
Execute: sudo dnf upgrade --enablerepo=qubes-vm-*-current-testing
Wait



### Expected behavior
No errors




### Actual behavior
Error shown in traceback above.


Notably, this also happened in my fedora-33 template as well.

**Comment by marmarek** [comment]
This looks rather similar to https://bugzilla.redhat.com/show_bug.cgi?id=1825213. The solution may be similar - checking `item.action`(?)  before accessing `item.pkg`.

**Comment by brendanhoar** [comment]
I see this issue is closed (via fixes for 4.1) but it was opened under 4.0. Will it be backported?

**Comment by heinrich-ulbricht** [comment]
Ran into this recently after upgrading to Fedora 34, found this ticket. What I'm wondering: is this error something to be worried about? Will the template degrade over time if this is not fixed?
